docs/tech_specs/configuration_backend_approach.md

```diff
@@ -0,0 +1,107 @@
+
+Overall, drawbacks of an umbrella project depends largely on our specific goals and needs. Umbrella projects seem to encourage more code sharing and therefore seem to be better suited for when you want to manage multiple, closely integrated applications. Moving things to an umbrella project may also induce an unnecessary initial learning curve as developers will need to learn about and adjust to working with the new structure.
+
+Moving/converting one or both projects for the umbrella may also cause us to lose the git history of either Screenplay or Screens.
+
+### References
+
+https://elixir-lang.org/getting-started/mix-otp/dependencies-and-umbrella-projects.html
+
+## 2. Library Dependency
+
+### Description
+
+The library approach would allow code to be shared between the two applications while keeping the codebases separate. The shared code would be separated out into its own library application and published as a Hex package for Screenplay to add as dependency and use.
+
+### High-level steps to get there
+
+- Create new library application
+- Move shared code into library
+- Publish to Hex
+- Add Hex package as a dependency in Screens and Screenplay’s `mix.exs` files
+- Fetch dependencies and start referencing Screens code
+
+### Benefits
+
+Setting up the shared code as a separate library and importing it as a package into Screenplay is a less intrusive solution since Screens and Screenplay will remain as separate projects. It also encourages a more modular structure and clearer separation of concerns. This approach is also arguably a more intuitive way of sharing code between projects and would therefore be less likely to be disruptive to the team and it’s workflow. A library also offers more flexibility since upgrades can be managed independently and an update to the shared code won’t trigger an automatic update to all repos to reference it.
+
+### Drawbacks
+
+While a library does enable greater flexibility in terms of version management, it also requires extra effort to do the management.
+
+### References
+
+https://hexdocs.pm/elixir/main/library-guidelines.html
+
+## 3. Git Dependency
+
+### Description
+
+Elixir also supports adding a GitHub repository directly as a dependency to your project. You can also pin the dependency to specific tags/commits/branches as well.
+
+### High-level steps to get there
+
+- Add the git dependency to the deps in Screenplay’s `mix.exs` file
+- Fetch dependencies and start referencing Screens code
+
+### Benefits
+
+This method is the least intrusive and easiest as it requires a trivial amount of upfront work to enable Screenplay to reference the code from Screens. It also requires no changes to the Screens application since we don’t need to move the shared code anywhere. We also have some of the same flexibility as the library approach since we can pin Screenplay to a specific version of Screens.
```

Comment:
I think we might run into the same problem here as I mentioned above—if the only way to access some Screens code is to get the entire application as a dependency, then we risk running into problems with having two different Phoenix web apps.

The git dependency approach could be useful for option 2 though—we could still split out the config code into a separate library, and skip the step of publishing it as an official package on hex, instead referencing the git repo in Screens/Screenplay deps.

Reply:
One more thing supporting the git-based dep idea: the [getting started guide](https://elixir-lang.org/getting-started/mix-otp/dependencies-and-umbrella-projects.html#internal-dependencies) differentiates "external" and "internal" dependencies.

External deps are for domain-agnostic libraries and such that your application essentially uses as tools.

Internal deps are for things that are specific to your project and just need to be in a separate package for some reason—pretty much exactly our use case. The git-based approach is recommended for internal deps.

Reply:
Ohh hm yeah that mixed approach is interesting! Personally, I would be down to skip publishing to hex. In practice, this approach seems like it would be the same as the hex package besides the fact that version management will be slightly different.